src/lib.rs

```diff
@@ -1,3 +1,187 @@
+//! # General
+//!
+//! The goal of this crate is to provide a robust and crypto-secure NAT traversal for peer to peer
+//! connection. It assumes publicly reachable rendezvous servers are provided. The server code
+//! itself is in the crate too, so the crate can either be used to deploy a server or used for peer
+//! to peer client communication or both simultaneously - for e.g. if you run the server on a port
+//! forwarded endpoint, it will be publicly available for others to rendezvous while you could
+//! choose normal NAT traversal mechanisms to communicate with other peers.
+//!
+//! ## Endpoint Independent Mapping (EIM)
```

Comment:
It would be nice to have a couple of sentences mentioning there are different NAT types before going deeper into that.

Reply:
But EIM etc _are_ the diff NAT types @povilasb .. check around page-6 [here](https://tools.ietf.org/html/rfc4787)

Reply:
Ah never mind - i think you meant just add a line explaining the upcoming sections are the various types of NAT of something. I did that, pls see if OK.